x-pack/plugins/cases/public/components/cases_context/cases_context_ui.tsx

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { getCreateCaseFlyoutLazyNoProvider } from '../../methods';
+import { CasesContextState } from './cases_context_reducer';
+
+export const CasesContextUI = ({ state }: { state: CasesContextState }) => {
```

Comment:
I think a `React.memo` would save some re-renders to this component.

Reply:
In practice every change to the context provider will affect cases context ui. A react memo won't save much in this particular case since this is what we need to do, re-render if anything changes.